### PR TITLE
Implement HTTP action interface in controllers

### PR DESCRIPTION
**Pull request by konarshankar07**
### Description (*)

This PR will enhance the security for the controllers. As per Magento Architecure, To improve security and logistics we need to allow limiting Actions to processing only requests with certain HTTP methods and add those limitations to as many existing Actions as possible. There are many vulnerabilities caused by actions processing both GET and POST requests and thus allowing bypassing security validations like form key validation. Also limiting actions to processing only requests with certain methods would serve as self-documentation for Action classes and improve consistency of server side for client code and functional tests.
More Information:- https://github.com/magento/architecture/blob/514952e8883234140071ec70f3b696d8267d52d0/design-documents/allowed-http-methods-for-actions.md

### CODE REVIEW

File: AdobeIms/Controller/Adminhtml/User/Profile.php [hunk 1]
```diff
@@ -10,14 +10,15 @@
...
+use Magento\Framework\App\Action\HttpPostActionInterface;
 use Magento\Framework\Controller\ResultFactory;
 use Magento\Framework\Exception\NoSuchEntityException;
 use Psr\Log\LoggerInterface;
 
 /**
  * Get Adobe services user account action
  */
-class Profile extends Action
+class Profile extends Action implements HttpPostActionInterface
```
Comment (sivaschenko) [inline AdobeIms/Controller/Adminhtml/User/Profile.php]: This controller is used to retrieve data only, let's make it implement the Get interface and change the call in `.modman/adobe-stock-integration/AdobeIms/view/adminhtml/web/js/signIn.js`

File: AdobeStockAdminUi/Controller/Adminhtml/System/Config/TestConnection.php [hunk 2]
```diff
@@ -12,14 +12,15 @@
...
+use Magento\Framework\App\Action\HttpPostActionInterface;
 use Magento\Framework\Controller\Result\Json;
 use Magento\Framework\Controller\Result\JsonFactory;
 use Magento\Framework\Controller\ResultInterface;
 
 /**
  * Controller used for testing connection to Adobe Stock API from stores configuration
  */
-class TestConnection extends Action
+class TestConnection extends Action implements HttpPostActionInterface
```
Comment (sivaschenko) [inline AdobeStockAdminUi/Controller/Adminhtml/System/Config/TestConnection.php]: This operation does not change the system - let's change it to Get and update connection.js as well

File: AdobeStockImageAdminUi/Controller/Adminhtml/License/Confirmation.php [hunk 3]
```diff
@@ -10,14 +10,15 @@
...
+use Magento\Framework\App\Action\HttpPostActionInterface;
 use Magento\Framework\Controller\Result\Json;
 use Magento\Framework\Controller\ResultFactory;
 use Psr\Log\LoggerInterface;
 
 /**
  * Backend controller for retrieving license confirmation before purchasing an asset
  */
-class Confirmation extends Action
+class Confirmation extends Action implements HttpPostActionInterface
```
Comment (sivaschenko) [inline AdobeStockImageAdminUi/Controller/Adminhtml/License/Confirmation.php]: I'd change this operation to Get as well

File: AdobeStockImageAdminUi/Controller/Adminhtml/License/Quota.php [hunk 4]
```diff
@@ -10,13 +10,14 @@
...
 use Magento\Backend\App\Action;
+use Magento\Framework\App\Action\HttpPostActionInterface;
 use Magento\Framework\Controller\ResultFactory;
 use Psr\Log\LoggerInterface;
 
 /**
  * Backend controller for retrieving license quota for the current user
  */
-class Quota extends Action
+class Quota extends Action implements HttpPostActionInterface
```
Comment (sivaschenko) [inline AdobeStockImageAdminUi/Controller/Adminhtml/License/Quota.php]: GET request would suite this operation better

File: MediaGalleryUi/Controller/Adminhtml/Directories/GetTree.php [hunk 5]
```diff
@@ -16,7 +17,7 @@
 /**
  * Returns all available directories
  */
-class GetTree extends Action
+class GetTree extends Action implements HttpPostActionInterface
```
Comment (sivaschenko) [inline MediaGalleryUi/Controller/Adminhtml/Directories/GetTree.php]: GET

**Review by sivaschenko — CHANGES_REQUESTED** [review]
Hi @konarshankar07 thanks for the great initiative! This pull request provided an opportunity to review the requests we are using for specific actions and I think we have to perform the requests that do not change the system using the GET method, it would be good to update such calls accordingly. Please see my comments

**Comment by konarshankar07** [comment]
Hello @sivaschenko ,
Thank you for the code review and I think logout request also needs to change to GET request. Let me know your thoughts

**Comment by sivaschenko** [comment]
@konarshankar07 thanks for the updates! The logout request does change a system state (admin is no longer logged in, tokens cleared from db) so it should be a POST request

File: AdobeStockImageAdminUi/view/adminhtml/web/js/signIn.js [hunk 6]
```diff
@@ -157,7 +157,7 @@ define([
          */
         loadUserProfile: function () {
             $.ajax({
-                type: 'POST',
+                type: 'GET',
                 url: this.profileUrl,
                 data: {
```
Comment (sivaschenko) [inline AdobeStockImageAdminUi/view/adminhtml/web/js/signIn.js]: form_key is not required for GET requests

File: AdobeStockImageAdminUi/view/adminhtml/web/js/signIn.js [hunk 7]
```diff
@@ -126,7 +126,7 @@ define([
          */
         getUserQuota: function () {
             $.ajax({
-                type: 'POST',
+                type: 'GET',
                 url: this.quotaUrl,
                 data: {
                     'form_key': window.FORM_KEY
```
Comment (sivaschenko) [inline AdobeStockImageAdminUi/view/adminhtml/web/js/signIn.js]: form_key is not required for GET requests

File: MediaGalleryUi/view/adminhtml/web/js/directory/directoryTree.js [hunk 8]
```diff
@@ -82,7 +82,7 @@ define([
         getJsonTree: function () {
             $.ajax({
                 url: this.getDirectoryTreeUrl,
-                type: 'POST',
+                type: 'GET',
                 dataType: 'json',
                 data: {
                     'form_key': FORM_KEY
```
Comment (sivaschenko) [inline MediaGalleryUi/view/adminhtml/web/js/directory/directoryTree.js]: form_key is not required for GET requests

File: AdobeIms/view/adminhtml/web/js/signIn.js [hunk 9]
```diff
@@ -80,7 +80,7 @@ define([
          */
         loadUserProfile: function () {
             $.ajax({
-                type: 'POST',
+                type: 'GET',
                 url: this.profileUrl,
                 showLoader: true,
                 data: {
```
Comment (sivaschenko) [inline AdobeIms/view/adminhtml/web/js/signIn.js]: form_key is not required for GET requests

**Review by sivaschenko — CHANGES_REQUESTED** [review]
Thanks for the updates @konarshankar07 ! Can you please also remove the form key from the requests changed to GET

**Review by sivaschenko — DISMISSED** [review]

**Review by sivaschenko — APPROVED** [review]

**Pull request merged**